packages/blockchain-wallet-v4-frontend/src/modals/Swap/CoinSelection/index.tsx

```diff
@@ -85,8 +87,19 @@ class CoinSelection extends PureComponent<Props> {
     }
   }
 
+  checkCustodialEligibility = (
+    custodialEligibility: boolean,
+    account: SwapAccountType
+  ) => {
+    if (account.type === 'CUSTODIAL' && !custodialEligibility) {
+      return false
+    } else {
+      return true
+    }
```

Comment:
this could be simplified to just 
`return !(account.type === 'CUSTODIAL' && !custodialEligibility)`